components/core/src/clp/ffi/SchemaTreeNode.hpp

```diff
@@ -0,0 +1,92 @@
+#ifndef CLP_FFI_SCHEMATREENODE_HPP
+#define CLP_FFI_SCHEMATREENODE_HPP
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace clp::ffi {
+/**
+ * This class implements a node in the schema tree. It stores information of a tree node, including
+ * the node type, key name, parent id, and ids of all the child nodes.
+ */
+class SchemaTreeNode {
+public:
+    // Types
+    using id_t = size_t;
```

Comment:
If this is designed to be serialized, then we should use a type that has an explicit size. What was our decision about the size of this type? If we didn't make one, perhaps @gibber9809 @wraymo, you can share some insights.

Reply:
In `clp-s` it's `int32_t`. Since we're not using negative ids, I will use `uint32_t` for now